src/components/progress-bar.tsx

```diff
@@ -27,26 +25,39 @@ const ProgressBar = ( {
 
 export default ProgressBar;
 
+type ProgressBarWithAutoIncrementProps = {
+	maxValue: number;
+	value: number;
+	increment?: number;
+};
+
 export function ProgressBarWithAutoIncrement( {
 	maxValue,
-	initialValue,
+	value,
 	increment = 2,
-}: {
-	maxValue: number;
-	initialValue: number;
-	increment?: number;
-} ) {
-	const [ value, setValue ] = useState( initialValue );
+}: ProgressBarWithAutoIncrementProps ) {
+	const [ animatedValue, setAnimatedValue ] = useState( value );
 
 	useEffect( () => {
+		if ( value > animatedValue ) {
+			setAnimatedValue( value );
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Let's add some description, otherwise aneother A12N can make mistake with adding it `animatedValue`. And it will save time for aoptehr A12S with debugging to understand teh reason and logic.